### PR TITLE
add E2E test setup

### PR DESCRIPTION
**Pull request by azmy60**
Hopefully this is enough for a basic E2E test setup https://github.com/thetarnav/solid-devtools/issues/223. I've added just a couple of tests as well. Let me know if there are things that need to change or add. This is my first time setting up playwright and GitHub workflow :innocent:

### CODE REVIEW

**Review by thetarnav — COMMENTED** [review]
wow this is incredible 😄
e2e tests would be something I'd be postponing forever, so I really appreciate it.
Just to have something running and checking if the debugger and app is loaded and interactive is a big win.
I'm surprised that you went for testing the actual extension. I was thinking of testing the overlay as I imagine it would be easier to maintain, and the functionality is almost identical (or should be) to extension.
But since the extension is working that's probably even better.
How hard do you think it would be to have two configs but reuse the tests to test both overlay and the extension? I'm guessing that's very possible.

I appreciate accessibility improvements too :)

**Comment by azmy60** [comment]
I'm glad you like it! Setting up the devtools was the trickiest one since there was really no documentation about it, and I had to do the research from both playwright & puppeteer repo, it was quite trial and errors.

And I think it wouldn't be hard to have both overlay and the extension configs running. I'll update you once I finish.

File: package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
         "sandbox:ext": "pnpm -dir examples/sandbox run dev:ext",
         "build": "turbo run build --filter=./packages/*",
         "test": "turbo run test --filter=./packages/*",
+        "test:e2e": "PW_CHROMIUM_ATTACH_TO_OTHER=1 playwright test -c playwright.config.ts",
```
Comment (thetarnav) [inline package.json:16]: This won't work on windows
Better assign it in the config with `process.env['PW_CHROMIUM_ATTACH_TO_OTHER'] = '1'` or use `cross-env`

**Comment by thetarnav** [comment]
I'm having some obscure errors while trying to run the tests locally. I have a feeling this is not gonna be fun debugging.. 😅

**Comment by azmy60** [comment]
Yeah, I just noticed some of the tests are flaky. :disappointed:

**Comment by azmy60** [comment]
@thetarnav I've added a little patch. I ran it 100 times and looks good in my local. Could you test it on your local again 😅? Hopefully it's fixed by now.

**Comment by thetarnav** [comment]
Ok I've tried this again, switched to linux in the meantime, and everything works great.
Thank you again for helping me with this and sorry for the delay.
Playwright looks like a really nice tool to use once it actually works 😊

**Pull request merged**